client/etcd/options.go

```diff
@@ -159,17 +166,26 @@ func NewOptions() Options {
 	return options{
 		sdOpts: services.NewOptions(),
 		iopts:  instrument.NewOptions(),
+		// NB(r): Set some default retry options so changes to retry
+		// option defaults don't change behavior of this client's retry options
+		retryOpts: retry.NewOptions().
+			SetInitialBackoff(defaultRetryInitialBackoff).
+			SetBackoffFactor(defaultRetryBackoffFactor).
+			SetMaxBackoff(defaultRetryMaxBackoff).
+			SetMaxRetries(defaultRetryMaxRetries).
+			SetJitter(defaultRetryJitter),
```

Comment:
Do we want to retry forever?

Reply:
Nah, crash looping the process after 16s is fine.  We just don't want to crash loop the process with no backoff at all.